v4/export/consistency.go

```diff
@@ -47,34 +54,33 @@ func (c *ConsistencyNone) TearDown() error {
 }
 
 type ConsistencyFlushTableWithReadLock struct {
+	ctx        context.Context
 	serverType ServerType
-	db         *sql.DB
+	conn       *sql.Conn
 }
 
 func (c *ConsistencyFlushTableWithReadLock) Setup() error {
 	if c.serverType == ServerTypeTiDB {
 		return withStack(errors.New("'flush table with read lock' cannot be used to ensure the consistency in TiDB"))
 	}
-	return FlushTableWithReadLock(c.db)
+	return FlushTableWithReadLock(c.ctx, c.conn)
 }
 
 func (c *ConsistencyFlushTableWithReadLock) TearDown() error {
-	err := c.db.Ping()
-	if err != nil {
-		return withStack(errors.New("ConsistencyFlushTableWithReadLock lost database connection"))
-	}
-	return UnlockTables(c.db)
+	defer c.conn.Close()
+	return UnlockTables(c.ctx, c.conn)
 }
 
 type ConsistencyLockDumpingTables struct {
-	db        *sql.DB
+	ctx       context.Context
```

Comment:
Do not store Contexts inside a struct type.

https://golang.org/pkg/context/#:~:text=Do%20not%20store%20Contexts%20inside,first%20parameter%2C%20typically%20named%20ctx%3A&text=See%20https%3A%2F%2Fblog.golang,a%20server%20that%20uses%20Contexts.